documentation regarding miniforge and fast version agnostic python p…

pnsinha opened the pull request:
…ackage manager uv

od101 commented on docs/software/apps-and-envs/python.md:
````diff
@@ -1,75 +1,300 @@
+=== "Midway3"
+    ```bash
+    source /project/PI_NAME/USER/envs/myenv/bin/activate
+    ```
````
formatting of above part 2 needs to be corrected.

od101 commented on docs/software/apps-and-envs/python.md:
````diff
@@ -1,75 +1,300 @@
+
+```bash
+# For conda environments
+conda env export --from-history > environment.yml
````
do you want to mention about tracking the channels?

od101 commented on docs/software/apps-and-envs/python.md:
```diff
@@ -1,75 +1,300 @@
+conda env export --from-history > environment.yml
+
+# For uv environments
+uv pip freeze > requirements.txt
```
maybe we can point to URL of docs as well? or perhaps once somewhere up when we started mentioning the `uv` the first time. (`uv --help` is not sufficient.)

od101 commented on docs/software/apps-and-envs/python.md:
````diff
@@ -1,75 +1,300 @@
+=== "Midway2"
+    ```bash
+    # Set default env location
+    export CONDA_ENVS_PATH=/project2/PI_NAME/USER/envs
````
maybe we've to mention the `CONDA_ENVS_DIRS` now?

od101 commented on docs/software/apps-and-envs/python.md:
````diff
@@ -1,75 +1,300 @@
+    # Set default env location
+    export CONDA_ENVS_PATH=/project/PI_NAME/USER/envs
+    ```
+
````
formatting needs to be corrected.

od101 commented on docs/software/apps-and-envs/python.md:
````diff
@@ -1,75 +1,300 @@
+    # Create read-only group environment
+    conda create --prefix=/project/PI_NAME/shared_envs/analysis python=3.11
+    chmod -R a-w /project/PI_NAME/shared_envs/analysis
+    ```
````
needs to be formatted correctly.

pnsinha replied:
I feel the formatting issue is due to tabbing. The pymdownx.tabbed extension allows to create tabs for organizing content. All code samples are presented in a tabbed interface for Midway2 and Midway3. If it is not rendered properly, it will look odd. Take a look at after running    "mkdocs serve".

vadim-cherepanov replied:
You are correct. When I run it with `mkdocs serve` rather than Github, I got nice blocks and tabs. There is still an issue with some numbered lists.

<img width="592" alt="Screenshot 2025-05-30 at 6 12 01 PM" src="https://github.com/user-attachments/assets/452c0321-6145-4daf-b91f-ee564de35cd1" />

This is how it renders for me in Safari with local `mkdocs serve`.

pnsinha commented on docs/software/apps-and-envs/python.md:
```diff
@@ -1,75 +1,300 @@
+
+2. **For Scientific Computing**: Use Miniforge (e.g., `python/miniforge-25.3.0`)
+   - Recommended conda distribution for research
+   - Uses conda-forge channel by default
```
added a line about conda-forge

pnsinha commented on docs/software/apps-and-envs/python.md:
```diff
@@ -1,75 +1,300 @@
+
+1. **For Standard Python**: Use version-specific modules
+   - On Midway3: (e.g., `python/3.11.9`)
+   - On Midway2: (e.g., `python/3.9.18`)
```
Now added current list for both clusters

pnsinha commented on docs/software/apps-and-envs/python.md:
```diff
@@ -1,75 +1,300 @@
+   - Uses conda-forge channel by default
+   - Smaller footprint than Anaconda
+   - Includes mamba for faster package resolution
+   - Available on both Midway2 and Midway3
```
Yes its about Miniforge.

pnsinha commented on docs/software/apps-and-envs/python.md:
```diff
@@ -1,75 +1,300 @@
 
-The command `module load python` will load the default module: an Anaconda distribution
-of Python. Note that there are multiple different Anaconda distributions available.  
+uv is a modern, fast alternative to pip for package management (available on both Midway2 and Midway3):
```
Installed earlier this month, but now the module is now also available.

pnsinha commented:
@od101 @vadim-cherepanov, do you have any suggestions for the introduction section? For instance, should we highlight the main audiences, like researchers and students, give a quick overview of the workflows we'll cover? Also, are there any common pain points or questions from users that you think we should address right away?

pnsinha commented:
**Unify and update Python documentation for Midway2 and Midway3**

1. Replaced previous Anaconda-centric documentation with a unified, modern guide for both Midway2 and Midway3 clusters.
2. Incorporated the latest best practices from the HPSF 2025 conference, with a focus on reproducibility, environment management, and efficient package handling.
3. Clarified and expanded instructions for using Miniforge, mamba, and uv, including rationale for their use over Anaconda.
4. Added or refined tables for domain-specific environments, and improved guidance on storage/cache management and project space usage.
5. Enhanced warnings, troubleshooting, and best-practice callouts for user clarity and reliability.
6. Note: This update replaces the previous documentation merged in commits d154ffeff8fb74bd96cb4f3c996853515b4e0d83 and 697d52024f647699b3687bfe7f6acca68d2a6a50, modernizing the core guidance and best practices for Python on RCC clusters. All changes are aimed at providing a consistent, community-driven, and reproducible Python experience for RCC users, while retaining essential advice for legacy Anaconda users.
7. These changes ensure all users benefit from the most current recommendations and a consistent experience across RCC clusters, reflecting both internal feedback and the latest HPC community standards.
8. As this page grows, we can plan to divide the Python documentation into multiple focused sections.

pnsinha commented:
**Reviewer Note:**
As this documentation uses MkDocs Material extensions (such as tabs, admonitions, and collapsible sections). To see all enhanced features rendered correctly, please use mkdocs serve to preview the site locally. Markdown renderers in VSCode or GitHub may not display these extensions as intended.
<img width="873" alt="Screenshot 2025-06-01 at 10 51 49 AM" src="https://github.com/user-attachments/assets/d60f1aad-a79b-478b-ac88-77e9d0b7c6ec" />
